### PR TITLE
add labels for servicemonitor

### LINKED ISSUE #298: be able to set labels for ServiceMonitor
## Overview
In my prometheus-operator configuration, I rely on the labels values in ServiceMonitor, so I need to be able to add my values to `labels`.

I think this is a popular practice and may be useful to others as well

## Solution
add a field to the template

### PR DESCRIPTION
**Pull request by batazor**
## Related issues/PRs: #298

Resolved #298

### CODE REVIEW

File: charts/kubescape-operator/templates/kubescape/servicemonitor.yaml [hunk 1]
```diff
@@ -8,6 +8,9 @@ metadata:
   labels:
     app: {{ .Values.kubescape.name }}
     helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    {{- with .Values.kubescape.serviceMonitor.customLabels }}
```
Comment (dwertent) [inline charts/kubescape-operator/templates/kubescape/servicemonitor.yaml:11]: I think we should have "labels" and not "customLabels".
What do you think?

**Review by dwertent — CHANGES_REQUESTED** [review]

File: charts/kubescape-operator/templates/kubescape/servicemonitor.yaml (hunk 1 above)
Comment (batazor) [reply charts/kubescape-operator/templates/kubescape/servicemonitor.yaml:11]: Yes, quite.

File: charts/kubescape-operator/templates/kubescape/servicemonitor.yaml [hunk 2]
```diff
@@ -8,6 +8,9 @@ metadata:
   labels:
     app: {{ .Values.kubescape.name }}
     helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    {{- with .Values.kubescape.serviceMonitor.labels }}
```
Comment (dwertent) [inline charts/kubescape-operator/templates/kubescape/servicemonitor.yaml:11]:
```suggestion
    {{- with .Values.kubescape.labels }}
```

File: charts/kubescape-operator/README.md [hunk 3]
```diff
@@ -155,6 +155,7 @@ docker-compose logs uptrace
 | kubescape.nodeSelector | object | `{}` | [Node selector](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/) |
 | kubescape.serviceMonitor.enabled | bool | `false` | enable/disable service monitor for prometheus (operator) integration |
 | kubescape.skipUpdateCheck | bool | `false` | skip check for a newer version |
+| kubescape.serviceMonitor.labels | `[]` | adds labels for servicemonitor |
```
Comment (dwertent) [inline charts/kubescape-operator/README.md:158]:
```suggestion
| kubescape.labels | `[]` | adds labels to the kubescape microservice |
```

**Review by dwertent — CHANGES_REQUESTED** [review]

File: charts/kubescape-operator/templates/kubescape/servicemonitor.yaml (hunk 2 above)
Comment (batazor) [reply charts/kubescape-operator/templates/kubescape/servicemonitor.yaml:11]: These are labels specifically for serviceMonitor, as they are only needed for prometheus-operator to work correctly. It seems to me that common labels will not work here

We can add separately `labels` or `commonLabels` for common cases.

Comment (batazor) [reply charts/kubescape-operator/templates/kubescape/servicemonitor.yaml:11]: @dwertent What do you think?

**Review by dwertent — APPROVED** [review]
Thanks :)

**Pull request merged**